agent/connect/csr_test.go

```diff
@@ -0,0 +1,32 @@
+package connect
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestCreateCSR_FormatDNSSANs(t *testing.T) {
+	pk, _, err := GeneratePrivateKey()
+	require.NoError(t, err)
+	spiffeID := &SpiffeIDService{
+		Host:       "7528f42f-92e5-4db4-b84c-3405c3ca91e6",
+		Service:    "srv1",
+		Datacenter: "dc1",
+	}
+	csr, err := CreateCSR(spiffeID, pk, []string{
+		"foo.example.com",
+		"foo.example.com:8080",
+		"bar.example.com",
```

Comment:
Maybe throw in a wildcard hostname in this test, like `*.example.com` because [that is allowed](https://developer.hashicorp.com/consul/docs/connect/config-entries/ingress-gateway#hosts)